diff --git a/main.go b/main.go
--- a/main.go
+++ b/main.go
@@ -384,25 +384,26 @@
 			}
 			// Network encoder
 			logger("Attempting network tune for device %s %s %v %v", t.url, t.tunerip, channel, idx)
+			if err := execute(t.pre, t.tunerip, channel); err != nil {
+				logger("[ERR] Failed to run pre script: %v %s", err, t.tunerip)
+				t.active = false
+				continue
+			}
 			resp, err := http.Get(t.url)
 			if err != nil {
 				logger("[ERR] Failed to fetch source: %v", err)
 				t.active = false
 				continue
 			} else if resp.StatusCode != 200 {
 				logger("[ERR] Failed to fetch source: %v", resp.Status)
+				resp.Body.Close()
 				t.active = false
 				continue
 			}
-			if err := execute(t.pre, t.tunerip, channel); err != nil {
-				logger("[ERR] Failed to run pre script: %v %s", err, t.tunerip)
-				t.active = false
-				continue
-			}
 			t.active = true
 			t.index = i
 			return &reader{
-				ReadCloser: resp.Body,
+				ReadCloser: WrapEncoderBody(resp.Body, t.url, t.tunerip),
 				channel:    channel,
 				t:          t,
 			}, nil
